aviator/aviator_test.go

```
@@ -175,11 +176,99 @@ func Test_DependencyCrawlingHarder(t *testing.T) {
 		"3": service3,
 		"4": service4,
 	}
-	err := a.resolveEnvironment("1")
+	err := a.resolveEnvironment("1", "")
 	assert.Nil(t, err)
 	assertOneOf(t, []string{"[4, 3, 2, 1]", "[4, 2, 3, 1]"}, a.environment.Services.String())
 }
 
+func Test_PromptUserConfig(t *testing.T) {
+	service := getTestService()
+	service.Name = "multi"
+
+	// Get rid of "default" to not confuse things.
+	delete(service.Setups, "default")
+
+	servicesetup := service.Setups["first"]
+	servicesetup.Command = "ls"
+	service.Setups["first"] = servicesetup
+
+	servicesetup = service.Setups["second"]
+	servicesetup.Command = "echo"
+	servicesetup.Default = true
+	service.Setups["second"] = servicesetup
+
+	a := New([]string{})
+	a.services = map[string]Service{
+		"multi": service,
+	}
+
+	var config ServiceSetup
+	var input bytes.Buffer
+
+	input.WriteString("second")
+	service.selectedSetup = ""
+	config = a.promptOrUseConfigForSetup(&service, "", &input)
+	assert.Equal(t, config.Command, "echo", "incorrect second command")
+
+	input.WriteString("first")
+	service.selectedSetup = ""
+	config = a.promptOrUseConfigForSetup(&service, "", &input)
+	assert.Equal(t, config.Command, "ls", "incorrect first command")
+
+	input.WriteString("")
+	service.selectedSetup = ""
+	config = a.promptOrUseConfigForSetup(&service, "", &input)
+	assert.Equal(t, config.Command, "echo", "incorrect default command")
+
+	input.WriteString("this_should_fail")
+	service.selectedSetup = ""
+	config = a.promptOrUseConfigForSetup(&service, "", &input)
+	assert.Equal(t, config.Command, "", "this should fail")
```

Sorry, yes. This was meant as a placeholder until we'd decided on how to handle an invalid config, name and if we even wanted to fail fast there.